tests/data/test987

```diff
@@ -0,0 +1,51 @@
+<testcase>
+<info>
+<keywords>
+SMTPS
+</keywords>
+</info>
+
+#
+# Server-side
+<reply>
+</reply>
+
+#
+# Client-side
+<client>
+<server>
+smtps
+</server>
+ <name>
+SMTPS with redundant explicit SSL request
+ </name>
+<stdin>
+From: different
+To: another
+
+body
+</stdin>
+ <command>
+-k --ssl-reqd --mail-rcpt recipient@example.com --mail-from sender@example.com -T - smtps://%HOSTIP:%SMTPSPORT/%TESTNUMBER
```

Comment:
I think these three tests should have `--ssl-reqd` removed. This option should only be necessary for transfers that are upgraded to TLS, while this one should use it from the start already (when using the `smtps://` URL scheme) and will fail if it can't do the TLS handshake.

Reply:
> I think these three tests should have --ssl-reqd removed. This option should only be necessary for transfers that are upgraded to TLS, 

No, this is intentional: see the test names. `--ssl-reqd` (or simply `--ssl`) should simply be ignored for implicit TLS protocols. Issue #10053 caused the TLS upgrade even for implicit TLS protocols and the goal of these tests is to check the upgrade is not attempted.

Regarding the CI test failures: I can't reproduce them locally thus it's hard to see what happens in the cloud. Obviouly it's a bad port reuse problem which seems to come from concurrent jobs. The server checks you mention above deal with pids, not ports. I wonder if checking for the port to be open can clear this problem.
Maybe it should also be good to wait for the ports to be closed when a server process is terminated, to avoid `EADDRINUSE` errors. I noticed  that, among random failures in CI jobs, many of them come from `EADDRINUSE`, even outside this PR: if we can resolve this, all CI tests will benefit of it.
I'm currently quite busy for a few days, so don't expect much activity on this PR soon.